### PR TITLE
Project hardening

### PR DESCRIPTION
**Pull request by bgannin**
Turns on additional compiler warnings and analyzer checks per #1214. Shallow analyzer is enabled by default now.

Addresses all reported issues so the project builds with the same number of deprecation warnings and passes both shallow and deep analysis phases. The specific changes are enumerated in the individual commits.

### CODE REVIEW

File: ResearchKit/Common/ORKWeightPicker.m [hunk 1]
```diff
@@ -322,10 +322,11 @@ - (NSArray *)kilogramValues {
     _canonicalMinimumValue = minimumValue;
     _canonicalMaximumValue = maximumValue;
     
-    for (double i = minimumValue; i <= maximumValue; i++) {
+    for (NSUInteger i = minimumValue; i <= maximumValue; i++) {
         [mutableWholeValues addObject:@(i)];
```
Comment (srinathtm-apple) [inline ResearchKit/Common/ORKWeightPicker.m]: since minimumValue is a double, I think it's risky to declare "i" as a NSUInteger...

Comment (bgannin) [reply ResearchKit/Common/ORKWeightPicker.m]: Iteration on float values is flagged by the compiler under security IIRC, it needs to be converted to a safer scalar format.

Comment (bgannin) [reply ResearchKit/Common/ORKWeightPicker.m]: Yep: CLANG_ANALYZER_SECURITY_FLOATLOOPCOUNTER. Solid discussion here: https://wiki.sei.cmu.edu/confluence/display/c/FLP30-C.+Do+not+use+floating-point+variables+as+loop+counters

Comment (srinathtm-apple) [reply ResearchKit/Common/ORKWeightPicker.m]: Ah thanks for sharing that article!

**Pull request merged**